tests/console/generate_dud.pm

```diff
@@ -13,24 +13,78 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
+use XML::Writer;
 use utils qw(zypper_call);
 use autoyast qw(expand_variables);
 use registration qw(add_suseconnect_product get_addon_fullname);
 
+sub generate_xml {
+    my $addons = get_var('BASE_TEST_REPOS', '');
+
+    # Generate addon products xml file
+    my $writer = XML::Writer->new(
+        DATA_MODE => 'true',
+        DATA_INDENT => 2,
+        OUTPUT => "self"
+    );
+    $writer->startTag(
+        "add_on_products",
+        xmlns => "http://www.suse.com/1.0/yast2ns",
+        "xmlns:config" => "http://www.suse.com/1.0/configns"
+    );
+    $writer->startTag("product_items", "config:type" => "list");
+    for my $addon (split(/,/, $addons)) {
+        my ($repo_id, $repo) = $addon =~ (/^\S+\/(\d+)\/(\S+)\/$/);
+        my $name = join '_', ($repo, $repo_id);
+        $writer->startTag("product_item");
+        $writer->startTag("url");
+        $writer->characters($addon);
+        $writer->endTag("url");
+        $writer->startTag("name");
+        $writer->characters($name);
+        $writer->endTag("name");
+        $writer->startTag("alias");
+        $writer->characters($name);
+        $writer->endTag("alias");
+        $writer->startTag("priority", "config:type" => "integer");
+        $writer->characters("50");
+        $writer->endTag("priority");
+        $writer->startTag("ask_user", "config:type" => "boolean");
+        $writer->characters("true");
+        $writer->endTag("ask_user");
+        $writer->startTag("selected", "config:type" => "boolean");
+        $writer->characters("true");
+        $writer->endTag("selected");
+        $writer->startTag("check_name", "config:type" => "boolean");
+        $writer->characters("true");
+        $writer->endTag("check_name");
+        $writer->endTag("product_item");
+    }
+    $writer->endTag("product_items");
+    $writer->endTag("add_on_products");
+    $writer->end();
+    return $writer->to_string;
+}
+
 sub run {
     select_console 'root-console';
 
     my $xml = 'add_on_products.xml';
     my $dud = get_required_var('DUD');
 
-    my $content = expand_variables(get_test_data($xml));
+    my $content =
+      (get_var('BASE_TEST_REPOS', '') eq '')
+      ? expand_variables(get_test_data($xml))
+      : generate_xml;
     save_tmp_file($xml, $content);
     add_suseconnect_product(get_addon_fullname('phub'));
     zypper_call('in mkdud');
     assert_script_run('wget -P inst-sys ' . autoinst_url . "/files/$xml");
+    script_run("cat inst-sys/$xml");
     assert_script_run("mkdud --create $dud --dist sle15 " .
           "--install instsys,repo --obs-keys --name 'Update' inst-sys");
     upload_asset($dud);
+    upload_logs("inst-sys/$xml");
```

Comment:
name of the xml should be `additional_products.xml` for example, to give a hint.

Reply:
This name was designed by original code. It uses "data/add_on_products.xml" to load xml file. So I cannot change name to others.

Reply:
it is just an internal name, renaming the file and the content of the variable should do.
but let's leave it, it is a minor thing.